beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestWithAccountBCDUpdate.java

```diff
@@ -175,11 +175,18 @@ private Account setupScenario() throws Exception {
         invoiceChecker.checkInvoice(invoices.get(3).getId(), callContext, expectedInvoices);
         expectedInvoices.clear();
 
-        // Move the BCD back to the 17 -- no repair happens upon the BCD change
+        // Move the BCD back to the 17 -- the 3 days between 2016-07-17 and 2016-07-20 are repaired
+        busHandler.pushExpectedEvents(NextEvent.INVOICE);
         accountUserApi.updateAccount(account.getId(), new DefaultMutableAccountData(null, null, null, 0, null, null, null, 17, null, null, null, null, null, null, null, null, null, null, null, null, null, false), callContext);
```

Comment:
Nit: One thought would be to enhance our `TestApiListener` and create a new `NextEvent.ACCOUNT_BCP_UPDATE` that we could expect here.

Reply:
I had looked into it -- this should really be `NextEvent.ACCOUNT_CHANGE` but that probably requires changing a lot of tests!